R/mod_spatial.R

```diff
@@ -0,0 +1,592 @@
+            c(
+              "basisOfRecord",
+              "kingdom",
+              "phylum",
+              "order",
+              "family",
+              "genus",
+              "species"
+            ),
+            selected = "basisOfRecord"
+          )
+        )
+      ),
+      column(class = "noPadding",
+             4,
+             plotlyOutput(
+               ns("records"),
+               height = "360px")
+      )
+    ),
+    fluidRow(
+      column(class = "noPadding",
+             12,
+             leafletOutput(
+               ns("mymap"),
+               height = "240px"
+             ),
+             absolutePanel(
+               top = 60,
+               right = 20,
+               selectInput(
+                 ns("mapTexture"),
+                 "Map Texture",
+                 choices = list(
+                   "OpenStreetMap.Mapnik" = "OpenStreetMap.Mapnik",
+                   "OpenStreetMap.BlackAndWhite" = "OpenStreetMap.BlackAndWhite",
+                   "Stamen.Toner" = "Stamen.Toner",
+                   "CartoDB.Positron" = "CartoDB.Positron",
+                   "Esri.NatGeoWorldMap" = "Esri.NatGeoWorldMap",
+                   "Stamen.Watercolor" = "Stamen.Watercolor",
+                   "Stamen.Terrain" = "Stamen.Terrain",
+                   "Esri.WorldImagery" = "Esri.WorldImagery",
+                   "Esri.WorldTerrain" = "Esri.WorldTerrain"
+                 ),
+                 selected = "Stamen.Toner"
+               ),
+               selectInput(
+                 ns("mapColor"),
+                 "Points Color",
+                 choices = list(
+                   "Red" = 'red',
+                   "Green" = "green",
+                   "Blue" = "blue",
+                   "Black" = "black"
+                 )
+               )
+             )
+      )
+    )
+  )
+}
+
+# Module Server
+
+#' @rdname mod_spatial
+#' @export
+#' @keywords internal
+mod_spatial_server <- function(input, output, session, data) {
+  ns <- session$ns
+  formattedData <- reactive({
+    dataForBar <- format_bdvis(data(), source = 'rgbif')
+    names(dataForBar) = gsub("\\.", "_", names(dataForBar))
+    if ("Date_collected" %in% colnames(dataForBar)) {
+      if (length(which(!is.na(dataForBar$Date_collected))) == 0) {
+        stop("Date_collected has no data")
+      }
+      dayofYear <- as.numeric(
+        strftime(
+          as.Date(
+            dataForBar$Date_collected,
+            na.rm = T
+          ),
+          format = "%d"
+        )
+      )
+      weekofYear <- as.numeric(
+        strftime(
+          as.Date(
+            dataForBar$Date_collected,
+            na.rm = T
+          ),
+          format = "%U"
+        )
+      )
+      monthofYear <- as.numeric(
+        strftime(
+          as.Date(
+            dataForBar$Date_collected,
+            na.rm = T
+          ),
+          format = "%m"
+        )
+      )
+      Year_ = as.numeric(
```

Comment:
Why are you naming this variable as Year_  ? any special reason for this format?

Reply:
Actually, I copied this line from here https://github.com/vijaybarve/bdvis/blob/master/R/tempolar.R#L72
But now I don't need this.